src/python/pants/backend/debian/rules.py

```diff
@@ -39,55 +43,58 @@ class DebianPackageFieldSet(PackageFieldSet):
 async def package_debian_package(
     field_set: DebianPackageFieldSet, tar_binary_path: TarBinary
 ) -> BuiltPackage:
-    dpkg_deb_path = await Get(
-        BinaryPaths,
+    dpkg_deb_path = await find_binary(
         BinaryPathRequest(
             binary_name="dpkg-deb",
             search_path=["/usr/bin"],
         ),
+        **implicitly(),
     )
     if not dpkg_deb_path.first_path:
         raise OSError(f"Could not find the `{dpkg_deb_path.binary_name}` program in `/usr/bin`.")
 
-    hydrated_sources = await Get(HydratedSources, HydrateSourcesRequest(field_set.sources_dir))
+    hydrated_sources = await hydrate_sources(
+        HydrateSourcesRequest(field_set.sources_dir), **implicitly()
+    )
 
     # Since all the sources are coming only from a single directory, it is
     # safe to pick an arbitrary file and get its root directory name.
     # Validation of the resolved files has been called on the target, so it is known that
     # snapshot.files isn't empty.
     sources_directory_name = PurePath(hydrated_sources.snapshot.files[0]).parts[0]
 
-    result = await Get(
-        ProcessResult,
-        Process(
-            argv=(
-                dpkg_deb_path.first_path.path,
-                "--build",
-                sources_directory_name,
-            ),
-            description="Create a Debian package from the produced packages.",
-            input_digest=hydrated_sources.snapshot.digest,
-            # dpkg-deb produces a file with the same name as the input directory
-            output_files=(f"{sources_directory_name}.deb",),
-            env={"PATH": str(PurePath(tar_binary_path.path).parent)},
-        ),
+    result = await fallible_to_exec_result_or_raise(
+        **implicitly(
+            Process(
+                argv=(
+                    dpkg_deb_path.first_path.path,
+                    "--build",
+                    sources_directory_name,
+                ),
+                description="Create a Debian package from the produced packages.",
+                input_digest=hydrated_sources.snapshot.digest,
+                # dpkg-deb produces a file with the same name as the input directory
+                output_files=(f"{sources_directory_name}.deb",),
+                env={"PATH": str(PurePath(tar_binary_path.path).parent)},
+            )
+        )
```

Comment:
wait what?!?

See if I get this right. We introduce call-by-name in order to get rid of the complicated rule graph building in the engine, but here we use `implicitly` to translate a `Process` into a `FallibleProcessResult` which would require us to still have that typed rule graph in place. 

Is this just a incremental step along the way? 
(I would've expected us to have to call something along the lines of: `await fallible_to_exec_result_or_raise(await process_request_to_process_result(Process(...)), ...)`.)

Reply:
Also, I find the name terrible to use. This will be the API surface for plugins to use, and this doesn't feel like a public API name to me.. BUT, this could of course be wrapped in layers later on, but I just feel like raising the question now, as this is the time if we want to make any naming adjustments without too much pain, I think.

Reply:
> as this is the time if we want to make any naming adjustments without too much pain

Disagree here, only because there will be EVEN MORE syntax/name fluff in the PRs. If we're going to re-name, I'd want to make sure we have aliases in place for a few revisions + a semi-automated migration (e.g. using libcst). That's a pretty simple migration compared to what we're doing - but adding it here would be yet more brutal.

> We introduce call-by-name in order to get rid of the complicated rule graph building in the engine, but here we use implicitly to translate a Process into a FallibleProcessResult which would require us to still have that typed rule graph in place.

Yeah, good point, I've been trying to keep my manual fiddling of the migration to a minimum - and basically enough to keep the compiler happy, and just removing the obvious stuff. Generally that has meant just going with whatever the migration spits out and not spending -too- much time re-jigging it by hand. Especially with nested async.

I don't actually understand why/how this is a rule - since, I thought it needed to be async?

```
@rule
def fallible_to_exec_result_or_raise(
    fallible_result: FallibleProcessResult,
    description: ProductDescription,
    keep_sandboxes: KeepSandboxes,
) -> ProcessResult:
```

Reply:
Oh wait, also, there are ambient parameters to pick up like `ProcessExecutionEnvironment` - so the implicitly will be there in some fashion or another 🤷🏽